backend-rust/migrations/0001_initialize.up.sql

```diff
@@ -0,0 +1,224 @@
+    'TransferWithSchedule',
+    'UpdateCredentials',
+    'RegisterData',
+    'ConfigureBaker',
+    'ConfigureDelegation'
+);
+
+CREATE TYPE credential_deployment_transaction_type AS ENUM (
+    'Initial',
+    'Normal'
+);
+
+CREATE TYPE update_transaction_type AS ENUM (
+    'UpdateProtocol',
+    'UpdateElectionDifficulty',
+    'UpdateEuroPerEnergy',
+    'UpdateMicroGtuPerEuro',
+    'UpdateFoundationAccount',
+    'UpdateMintDistribution',
+    'UpdateTransactionFeeDistribution',
+    'UpdateGasRewards',
+    'UpdateBakerStakeThreshold',
+    'UpdateAddAnonymityRevoker',
+    'UpdateAddIdentityProvider',
+    'UpdateRootKeys',
+    'UpdateLevel1Keys',
+    'UpdateLevel2Keys',
+    'UpdatePoolParameters',
+    'UpdateCooldownParameters',
+    'UpdateTimeParameters',
+    'MintDistributionCpv1Update',
+    'GasRewardsCpv2Update',
+    'TimeoutParametersUpdate',
+    'MinBlockTimeUpdate',
+    'BlockEnergyLimitUpdate',
+    'FinalizationCommitteeParametersUpdate'
+);
+
+CREATE TYPE transaction_type AS ENUM (
+    'Account',
+    'CredentialDeployment',
+    'Update'
+);
+
+
+-- Every block on chain.
+CREATE TABLE blocks(
+    -- The absolute height of the block.
+    height
+        BIGINT
+        PRIMARY KEY
+        NOT NULL,
+    -- Block hash encoded using HEX.
+    hash
+        CHAR(64)
+        UNIQUE
+        NOT NULL,
+    -- Timestamp for when the block was baked.
+    slot_time
+        TIMESTAMP
+        NOT NULL,
+    -- Milliseconds between the slot_time of this block and the block below (height - 1).
+    -- For the genesis block it will be 0.
+    block_time
+        INTEGER
+        NOT NULL,
+    -- Milliseconds between the slot_time of this block and the block above causing this block to be finalized.
+    -- This is NULL until the indexer have processed the block marking this a finalized.
+    finalization_time
+        INTEGER,
+    -- Block causing this block to become finalized.
+    -- This is NULL until the indexer have processed the block marking this a finalized.
+    finalized_by
+        BIGINT
+        REFERENCES blocks(height),
+    -- Index of the account which baked the block.
+    -- For non-genesis blocks this should always be defined.
+    -- Foreign key constraint added later, since account table is not defined yet.
+    baker_id
+        BIGINT,
+    -- The total amount of CCD in existence at the time of this block was created in micro CCD.
+    total_amount
+        BIGINT
+        NOT NULL,
+    -- The total staked amount of CCD at the time of this block was created in micro CCD.
+    total_staked
+        BIGINT
+        NOT NULL
+);
+
+-- Every transaction on chain.
+CREATE TABLE transactions(
+    -- Index of the transaction within the block.
+    index
+        BIGINT
+        NOT NULL,
+    -- Absolute height of the block containing the transaction.
+    block
+        BIGINT
+        REFERENCES blocks(height)
+        NOT NULL,
+    -- Transaction hash encoded using HEX.
+    hash
```

Comment:
Might not be included in this initial PR. However, adding indexes beyond the primary key index would help to increase the performance in common queries.

E.g. an index for querying transactions/bocks based on the transaction_hash/block_hash.

Reply:
Yes, I intend to do that, but I would like to have most of the implementation ready, before we start optimizing.
It would be good with some realistic numbers to understand the impact, although I agree that these cases will obviously benefit from it.